IBNet/TWSEncoding.cs

```diff
@@ -89,11 +173,29 @@ internal EnumEncDec(Type t)
     }
 
     const string IB_EXPIRY_DATE_FORMAT = "yyyyMMdd";
-    readonly string NUMBER_DECIMAL_SEPARATOR;
 
     static readonly Dictionary<Type, EnumEncDec> _enumDecoders;
 
-    protected Stream _stream;
+    protected Stream Stream;
+
+    int _inEncodingBatch;
+    protected bool InEncodingBatch
+    {
+      get => _inEncodingBatch != 0;
+      set => Interlocked.Exchange(ref _inEncodingBatch, value ? 1 : 0);
+    }
+
+    int _inDecodingBatch;
+    protected bool InDecodingBatch
+    {
+      get => _inEncodingBatch != 0;
+      set => Interlocked.Exchange(ref _inDecodingBatch, value ? 1 : 0);
```

Comment:
Why are you using Interlocked here?
IIRC, the setters are only called inside a lock.
What am I missing here?